src/App.tsx

```diff
@@ -1,223 +1,257 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  addTodo,
+  editTodo,
+  getTodos,
+  removeTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Errors } from './components/Errors';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './types/Errors';
+import { Status } from './types/Status';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState<boolean>(false);
+  const [currError, setCurrError] = useState<string>('');
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [isLoadingTodosIds, setIsLoadingTodosIds] = useState<number[]>([]);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+  const getTodosFromServer = useCallback(async () => {
+    try {
+      const todosFromServer = await getTodos(user?.id || 1);
+
+      setTodos(todosFromServer);
+    } catch {
+      setHasError(true);
+      setCurrError(ErrorMessage.NoTodos);
     }
   }, []);
```

Comment:
you can add user to array of deps so when user have changed there will be his todos

Reply:
also, do you really need this? you have identical function at 142 line